backend/src/federation/index.ts

```diff
@@ -36,18 +52,47 @@ export const startDHT = async (
     logger.info(`keyPairDHT: publicKey=${keyPair.publicKey.toString('hex')}`)
     logger.debug(`keyPairDHT: secretKey=${keyPair.secretKey.toString('hex')}`)
 
+    const apiList = prepareCommunityApiList()
+    logger.debug(`ApiList: ${JSON.stringify(apiList)}`)
+
     const node = new DHT({ keyPair })
 
     const server = node.createServer()
 
     server.on('connection', function (socket: any) {
       // noiseSocket is E2E between you and the other peer
       // pipe it somewhere like any duplex stream
-      logger.info(`Remote public key: ${socket.remotePublicKey.toString('hex')}`)
+      logger.info(`server on... with Remote public key: ${socket.remotePublicKey.toString('hex')}`)
       // console.log("Local public key", noiseSocket.publicKey.toString("hex")); // same as keyPair.publicKey
 
-      socket.on('data', (data: Buffer) => logger.info(`data: ${data.toString('ascii')}`))
+      socket.on('data', async (data: Buffer) => {
+        logger.info(`data: ${data.toString('ascii')}`)
+        const json = JSON.parse(data.toString('ascii'))
```

Comment:
you are right to say it is the attack vector here on receiving data on socket.
That's the reason to define the acceptance of data on a dedicated type of data like it is done directly after parsing the data as a JSON object and check for the array of "apiVersions" containing the attributes "api" and "url"

Sending the list of apiVersions per separate socket.writes will cause asynchronous and parallel invocations of socket.on and we have to synchronize the write access on the same database table. For sure it could be done by the database itself, but on my opinion avoiding proactive will be a better solution than resolving it afterwards...